typescript/cli/src/validator/address.ts

```diff
@@ -0,0 +1,145 @@
+import { GetPublicKeyCommand, KMSClient } from '@aws-sdk/client-kms';
+import { GetObjectCommand, S3Client } from '@aws-sdk/client-s3';
+// @ts-ignore
+import asn1 from 'asn1.js';
+import { ethers } from 'ethers';
+
+import { CommandContext } from '../context/types.js';
+import { log, logBlue } from '../logger.js';
+import { getAccessKeyId, getRegion, getSecretAccessKey } from '../utils/aws.js';
+
+export async function getValidatorAddress({
+  context,
+  accessKey,
+  secretKey,
+  region,
+  bucket,
+  keyId,
+}: {
+  context: CommandContext;
+  accessKey?: string;
+  secretKey?: string;
+  region?: string;
+  bucket?: string;
+  keyId?: string;
+}) {
+  if (!bucket && !keyId) {
+    throw new Error('Must provide either an S3 bucket or a KMS Key ID.');
+  }
+
+  // Query user for AWS parameters if not passed in or stored as .env variables
+  accessKey ||= await getAccessKeyId(context.skipConfirmation);
+  secretKey ||= await getSecretAccessKey(context.skipConfirmation);
+  region ||= await getRegion(context.skipConfirmation);
+
+  if (!accessKey) {
+    throw new Error('No access key ID set.');
+  }
+
+  if (!secretKey) {
+    throw new Error('No secret access key set.');
+  }
+
+  if (!region) {
+    throw new Error('No AWS region set.');
+  }
```

Comment:
Please import the `assert` helper function from the `@hyperlane-xyz/utils` package and shorten these checks to:
`assert(accessKey, 'No access key ID set.')`